.github/actions/package-windows-qt/action.yml

```diff
@@ -1,90 +1,122 @@
     required: true
   outputName:
-    description: 'Name pattern for created Windows artifacts'
+    description: Name pattern for created Windows artifact
+    required: true
+  pdbOutputName:
+    description: Name pattern for created Windows PDB artifact
     required: true
 runs:
-  using: 'composite'
+  using: composite
   steps:
     - name: Download Windows RelWithDebInfo artifact
       uses: actions/download-artifact@v3
       with:
-        name: ${{ inputs.base }}-RelWithDebInfo-${{ github.sha }}
-
-    - name: Get File Names
-      id: GetFileNames
-      shell: pwsh
-      run: |
-        $FileName = (Get-ChildItem -Filter "windows-deps-qt*.zip").Name.Replace("-RelWithDebInfo", "")
-        $PDBArchiveFileName = (Get-ChildItem -Filter "windows-deps-qt*.zip").Name.Replace("RelWithDebInfo", "ReleasePDBs")
-        $PDBOutputName = "${{ inputs.outputName }}".Replace("${{ inputs.base }}", "${{ inputs.base }}-ReleasePDBs")
-        "fileName=${FileName}" >> $env:GITHUB_OUTPUT
-        "pdbFileName=${PDBArchiveFileName}" >> $env:GITHUB_OUTPUT
-        "pdbOutputName=${PDBOutputName}" >> $env:GITHUB_OUTPUT
-
-    - name: Extract RelWithDebInfo
-      shell: pwsh
-      run: |
-        7z x 'windows-deps-qt*.zip' -oqt_rel
-        Remove-Item "*.zip"
-
-    - name: Separate RelWithDebInfo PDBs
-      shell: pwsh
-      run: |
-        New-Item -ItemType "directory" rel_pdbs
-        Set-Location "qt_rel"
-        $QtInstallDir = (Get-Location | Convert-Path)
-        Set-Location "..\rel_pdbs"
-        $ReleasePdbInstallDir = (Get-Location | Convert-Path)
-        Set-Location ".."
-        $ReleasePdbFiles = Get-ChildItem -Filter "*.pdb" -File -Recurse
-        $DestinationDirRelativePaths = ( $ReleasePdbFiles | ForEach-Object { $_.DirectoryName } | Sort-Object -Unique ).Replace("${QtInstallDir}\", "")
-        $DestinationDirRelativePaths | ForEach-Object { New-Item -Name "$_" -Path "${ReleasePdbInstallDir}" -ItemType "directory" -Force }
-        $ReleasePdbFiles | ForEach-Object { Move-Item $_.FullName $_.FullName.Replace("${QtInstallDir}", "${ReleasePdbInstallDir}") }
+        name: ${{ inputs.releaseArtifact }}
 
     - name: Download Windows Debug artifact
       uses: actions/download-artifact@v3
       with:
-        name: ${{ inputs.base }}-Debug-${{ github.sha }}
+        name: ${{ inputs.debugArtifact }}
 
-    - name: Extract Debug
+    - name: Combine Artifacts
+      id: combine
       shell: pwsh
       run: |
-        7z x 'windows-deps-qt*.zip' -oqt
-        Remove-Item "*.zip"
+        # Combine Artifacts
 
-    - name: Copy RelWithDebInfo over Debug
-      shell: pwsh
-      run: |
-        Remove-Item -Path "qt" -Include "cmake_automoc_parser.*","moc.*","qlalr.*","qmake.*","qsb.*","qtpaths.*","qvkgen.*","rcc.*","tracegen.*","uic.*" -Recurse
-        Copy-Item "qt_rel\*" "qt\" -Recurse -Force
+        . (Join-Path -Path $env:GITHUB_ACTION_PATH -ChildPath 'Invoke-External.ps1')
```

Comment:
I found it to be more "correct" than referring to a file somewhere else in the repository, as it would also make the action less "self-contained".

Reply:
Yeah, I understand the rationale. Something about it feels "off", but I can't put my finger on it, and it might just be irrational dislike. Don't worry about it.